controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -844,6 +937,27 @@ bool wfa_ca::get_send_1905_1_tlv_hex_list(std::list<tlv_hex_t> &tlv_hex_list,
     return false;
 }
 
+/**
+ * @brief Writes a mac string into the class buffer.
+ * 
+ * @param[in] value mac address.
+ * @param[out] length of current tlv.
+ * @return false if length smaller than data, else true.
+ */
+bool tlvPrefilledData::add_tlv_value_mac(std::string &value, uint16_t &length)
+{
+    for (size_t char_idx = 0; char_idx < value.size(); char_idx += 2) {
+        if (length == 0) {
+            return false;
+        }
+        *m_buff_ptr__ = std::stoi(value.substr(char_idx, 2), nullptr, 16);
+        m_buff_ptr__++;
+        length--;
+        char_idx++;
```

Comment:
It is really weird to update char_idx here *and* in the loop update. Is there a reason for that?

Reply:
You're right it was a temporary change, I'll change that